### PR TITLE
Bring imagec binary back for dev tests

### LINKED ISSUE #5330: Add imagec binary build back into VIC for dev purposes
Acceptance criteria:

- [ ] Update makefile so we can test the transfer component of docker push while we wait for docker diff to complete.

---

Docker push has a dependency on docker diff.  We cannot do the image construction part of push without diff.  If we reassembled the imagec binary, we can at least test docker push on a tagged image or at least work on that transfer code while we wait for docker diff to complete.

### PR DESCRIPTION
**Pull request by chengwang86**
fixes #5330 

This is only used for dev test purposes so there is no need to add integration test.
 
To run it in standalone mode without starting persona/portlayer, run `sudo bin/imagec --reference=busybox:latest --operation=push --standalone`; this cmd will pull `busybox` from dockerhub by default. 

In standalone mode, the image tar will be written to a local folder `images`

### CODE REVIEW

File: cmd/imagec/hook.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
```
Comment (sgairo) [inline cmd/imagec/hook.go]: 2017 :)

File: lib/imagec/imagec.go [hunk 2]
```diff
@@ -107,6 +106,9 @@ type Options struct {
 
 	// RegistryCAs will not be modified by imagec
 	RegistryCAs *x509.CertPool
+
+	// If being set to true, do not bother portlayer or persona
```
Comment (jzt) [inline lib/imagec/imagec.go]: `s/being set to//g`

**Comment by chengwang86** [comment]
From ci test:
```
building /go/bin/gas...
checking security problems
make: *** [gas] Error 1
Makefile:231: recipe for target 'gas' failed
```
I need to figure out where the `gas` check was violated.

**Comment by sflxn** [comment]
Can you remove it from most and other make targets?  This is just for development purposes so we should just build it if we call it out at the command line.

**Comment by sflxn** [comment]
lgtm.  just rebase and squash.

**Pull request merged**